tensorrt_llm/_torch/models/modeling_nanov2vlm.py

```diff
@@ -117,6 +120,47 @@ def extract_feature(self, pixel_values):
         vit_embeds = self.mlp1(vit_embeds)
         return vit_embeds
 
+    def apply_evs(
+            self, mm_embedding: List[torch.Tensor],
+            multimodal_params: List[MultimodalParams]) -> List[torch.Tensor]:
+        """Apply EVS to the multimodal embedding."""
+        if VIDEO_PRUNING_RATIO <= 0:
+            return mm_embedding
+
+        modality_types = [
+            multimodal_param.multimodal_data['modality_type']
+            for multimodal_param in multimodal_params
+        ]
+        video_size_list = [
+            multimodal_param.multimodal_data['video_size']
+            for multimodal_param in multimodal_params
+        ]
+        mm_embedding_evs = []
+        # Iterate over batch.
+        for modality_type, mm_embed, video_sizes in zip(modality_types,
+                                                        mm_embedding,
+                                                        video_size_list):
+            if modality_type == "video":
+                # Iterate over each video in the batch.
+                start_idx, mm_embed_list = 0, []
+                for video_size in video_sizes:
+                    partial_mm_embed = mm_embed[start_idx:start_idx +
+                                                video_size[0]]
+                    retention_mask = compute_retention_mask(
+                        video_embeds=partial_mm_embed,
+                        video_size=video_size,
+                        spatial_merge_size=self.spatial_merge_size,
+                        q=VIDEO_PRUNING_RATIO,
+                        flatten_output=False,
+                    ).flatten(start_dim=1)
```

Comment:
I'm not sure why you need to flatten here since `compute_retention_mask` operates on `[T, Hidden]` shape and returns mask of `[T]` shape. If you get different shape this probably indicate you're applying EVS to wrong input shape.